### PR TITLE
Move procspy out of vendor into probe/endpoint.

### PR DESCRIPTION
**Pull request by tomwilkie**
Part of #284

### CODE REVIEW

File: probe/endpoint/procspy/proc_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+				},
+			),
+		),
+	),
+)
+
+func TestWalkProcPid(t *testing.T) {
+	oldReadDir, oldLstat, oldStat, oldOpen := readDir, lstat, stat, open
+	defer func() { readDir, lstat, stat, open = oldReadDir, oldLstat, oldStat, oldOpen }()
+	readDir, lstat, stat, open = mockFS.ReadDir, mockFS.Lstat, mockFS.Stat, mockFS.Open
```
Comment (paulbellamy) [inline probe/endpoint/procspy/proc_internal_test.go]: Possible to wrap this up a bit? Your intention is being obscured by the stutter.
```Go
resetFS := fs.Mock(mockFS)
defer resetFS()
```

Comment (tomwilkie) [inline probe/endpoint/procspy/proc_internal_test.go]: Yeah I did that in the next PR - I can bring it forward it you like?

Comment (paulbellamy) [inline probe/endpoint/procspy/proc_internal_test.go]: Oh, wait these are all internal to this package, not from `fs`. That's a bit weird.

Comment (tomwilkie) [inline probe/endpoint/procspy/proc_internal_test.go]: Yeah, also fixed in the next PR.  Want me to bring it forward?

Comment (paulbellamy) [inline probe/endpoint/procspy/proc_internal_test.go]: ah, no that's fine then.

**Comment by paulbellamy** [comment]
LGTM

**Pull request merged**